CURA-7812-decrease-Qml-singletons

saumyaj3 opened the pull request:
#Description
CURA-7812

wawanbreton commented on resources/qml/Settings/SettingTextField.qml:
```diff
@@ -223,7 +223,7 @@ SettingItem
                 cursorShape: Qt.IBeamCursor
 
-                onPressed: {
+                onPressed:(mouse)=> {
```
I never saw this QML syntax :smile: what does it do ?

wawanbreton replied:
Oh yes indeed, because `mouse` is used implicitely inside the function, so that makes it explicit. Good to know :smiley: :+1:

wawanbreton reviewed (commented):
I'm a bit disturbed by the fact of adding deprecated functions, this is quite a non-sense :smile: However I understand that deprecating a singletonization is not really possible.
On the other hand, I see no warning pop up when I run your code. What is the actual effect of the `deprecated` decorator ? Is it just a reminder for ourselves ?

saumyaj3 commented:
> I'm a bit disturbed by the fact of adding deprecated functions, this is quite a non-sense 😄 However I understand that deprecating a singletonization is not really possible. On the other hand, I see no warning pop up when I run your code. What is the actual effect of the `deprecated` decorator ? Is it just a reminder for ourselves ?

If you go to qml and use one of the registersingletons, the warnings appear. At the moment I removed all the registerSingletons used in Cura. Replaced it with what we desire. They are not used in any part of our code. but you can use them with the @deprecated warning.

wawanbreton commented:
Ok, so the warning don't show up for us, but they will if someone is using code calling them, that's perfect :+1:
